### PR TITLE
Add support for reimbursement requests

### PR DESCRIPTION
**Pull request by ManfredKarrer**

### CODE REVIEW

**Comment by ManfredKarrer** [comment]
Implements https://github.com/bisq-network/proposals/issues/49

File: core/src/main/java/bisq/core/btc/wallet/BtcWalletService.java [hunk 1]
```diff
@@ -174,9 +191,11 @@ public Transaction completePreparedCompensationRequestTx(Coin issuanceAmount, Ad
         // BSQ change outputs from BSQ fee inputs.
         feeTx.getOutputs().forEach(preparedTx::addOutput);
 
-        // BSQ issuance output
-        preparedTx.addOutput(issuanceAmount, issuanceAddress);
-
+        //
```
Comment (sqrrm) [inline core/src/main/java/bisq/core/btc/wallet/BtcWalletService.java]:
```suggestion
        // For generic proposals there is no issuance output, for compensation and reimburse requests there is
```

File: core/src/main/java/bisq/core/dao/governance/voteresult/issuance/IssuanceService.java [hunk 2]
```diff
@@ -50,42 +55,51 @@ public IssuanceService(DaoStateService daoStateService, PeriodService periodServ
...
-                .filter(txOutput -> isValid(txOutput, compensationProposal, periodService, chainHeight))
+                .filter(txOutput -> isValid(txOutput, issuanceProposal, periodService, chainHeight))
                 .forEach(txOutput -> {
+                    IssuanceType issuanceType = IssuanceType.UNDEFINED;
+                    if (issuanceProposal instanceof CompensationProposal) {
+                        issuanceType = IssuanceType.COMPENSATION;
+                    } else if (issuanceProposal instanceof ReimbursementProposal) {
+                        issuanceType = IssuanceType.REIMBURSEMENT;
+                    }
+                    checkArgument(issuanceType != IssuanceType.UNDEFINED, "issuanceType must nto be undefined");
```
Comment (sqrrm) [inline core/src/main/java/bisq/core/dao/governance/voteresult/issuance/IssuanceService.java]:
```suggestion
                    checkArgument(issuanceType != IssuanceType.UNDEFINED, "issuanceType must not be undefined");
```

File: core/src/main/java/bisq/core/dao/state/governance/IssuanceType.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with Bisq. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package bisq.core.dao.state.governance;
+
+public enum IssuanceType {
```
Comment (sqrrm) [inline core/src/main/java/bisq/core/dao/state/governance/IssuanceType.java:20]: GENESIS could be a reasonable type as well, no need really.

**Review by sqrrm — APPROVED** [review]
utACK

File: core/src/main/java/bisq/core/dao/state/governance/IssuanceType.java (hunk 3 above)
Comment (ManfredKarrer) [reply core/src/main/java/bisq/core/dao/state/governance/IssuanceType.java:20]: Good point but as it is not treated as Issuance I would leave it out.

**Pull request merged**